course3/homework-string/my_string.cc

```diff
@@ -0,0 +1,113 @@
+#include "my_string.h"
+# include <iostream>
+
+MyString::MyString(){
+//	std::cout << "default constructor" << std::endl;
+};
+
+MyString::MyString(const char* inchar){
+//	std::cout << "string literal constructor called" << std::endl;
+	int n = std::strlen(inchar) + 1;
+	data = std::unique_ptr<char[]>(new char[n]);
+ 	std::strcpy(data.get(), inchar);
+};
+
+MyString::MyString(const MyString& that): data(new char[that.size()+1]){
+//	std::cout << "const ref copy constructor" << std::endl;
+	std::strcpy(data.get(), that.data.get());
+}
+
+
+bool MyString::empty() const{
+	return size() == 0;
+};
+
+int MyString::size() const{
+	if (data == 0)
```

Comment:
Please, don't compare pointer to int literal, this makes code confusing ("is 'data' an integer?). Either compare it to ```nullptr```, or just take advantage of implicit boolean conversion.